crates/pera-core/src/signature_mpc/mpc_manager.rs

```diff
@@ -79,33 +165,77 @@ pub struct SignatureMPCManager {
     pending_instances_queue: VecDeque<ObjectID>,
     // TODO (#257): Make sure the counter is always in sync with the number of active instances.
     active_instances_counter: usize,
+    language_public_parameters:
+        maurer::language::PublicParameters<{ maurer::SOUND_PROOFS_REPETITIONS }, Lang>,
+    consensus_adapter: Arc<dyn SubmitToConsensus>,
+    pub epoch_store: Weak<AuthorityPerEpochStore>,
+    pub max_active_mpc_instances: usize,
+    threshold: usize,
+}
+
+type Lang = maurer::knowledge_of_discrete_log::Language<secp256k1::Scalar, secp256k1::GroupElement>;
+type ProofParty = proof::aggregation::asynchronous::Party<
+    maurer::Proof<{ maurer::SOUND_PROOFS_REPETITIONS }, Lang, PhantomData<()>>,
+>;
+
+fn generate_language_public_parameters<const REPETITIONS: usize>(
+) -> maurer::language::PublicParameters<REPETITIONS, Lang> {
+    let secp256k1_scalar_public_parameters = secp256k1::scalar::PublicParameters::default();
+
+    let secp256k1_group_public_parameters = secp256k1::group_element::PublicParameters::default();
+
+    PublicParameters::new::<secp256k1::Scalar, secp256k1::GroupElement>(
+        secp256k1_scalar_public_parameters,
+        secp256k1_group_public_parameters.clone(),
+        secp256k1_group_public_parameters.generator,
+    )
 }
 
 impl SignatureMPCManager {
-    pub fn new() -> Self {
+    pub fn new(
+        consensus_adapter: Arc<dyn SubmitToConsensus>,
+        epoch_store_weak: Weak<AuthorityPerEpochStore>,
+        max_active_mpc_instances: usize,
+        num_of_parties: usize,
+    ) -> Self {
         Self {
             mpc_instances: HashMap::new(),
             pending_instances_queue: VecDeque::new(),
             active_instances_counter: 0,
+            language_public_parameters: generate_language_public_parameters::<
+                { maurer::SOUND_PROOFS_REPETITIONS },
+            >(),
+            consensus_adapter,
+            epoch_store: epoch_store_weak,
+            max_active_mpc_instances,
+            threshold: ((num_of_parties / 3) * 2) + 1,
```

Comment:
This formula is (slightly) wrong. Look at `q` in section 1.1 of the design document for the exact formula.

(P.S. had this been the correct formula, you should have multiplied by two first, and divided by three second. Right now, there is information loss because of the integer division :wink: Please pay attention to this when implementing the right formula!)

Reply:
Fixed